### ISSUE TITLE
luis-apis, luisgen, botdispatch, msbot do not install on macOS High Sierra

### ISSUE THREAD

**Issue by vishwacsena** [Microsoft/botbuilder-tools#392]
1. Connect to nightly feed - https://botbuilder.myget.org/F/botbuilder-tools-daily/npm/
2. sudo npm i -g luis-apis luisgen botdispatch msbot

Expected: All tools install successfully

Actual: Various errors. 

Note: chatdown, ludown, qnamaker do install fine and work well.

**Comment by enzocanoo** [comment]
Hi @vishwacsena, we only found problems with the `msbot` tools due to a bug in the publish script. Could you provide us the logs related to the `npm install` to a better understanding?

**Comment by vishwacsena** [comment]
@enzocanoo - sure. 

# LUIS
```bash
> sudo npm config set registry https://botbuilder.myget.org/F/botbuilder-tools-daily/npm/

> sudo npm i -g luis-apis
/usr/local/bin/luis -> /usr/local/lib/node_modules/luis-apis/bin/luis
+ luis-apis@2.0.10-5
added 111 packages from 208 contributors in 16.458s

> luis
module.js:549
    throw err;
    ^

Error: Cannot find module '../lib/luisAuthoring'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/luis-apis/bin/luis:12:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
```

# Luis Gen

```bash
> sudo npm config set registry https://botbuilder.myget.org/F/botbuilder-tools-daily/npm/

> sudo npm i -g luisgen
/usr/local/bin/luisgen -> /usr/local/lib/node_modules/luisgen/bin/luisgen

> dotnet-2.0.0@1.4.4 postinstall /usr/local/lib/node_modules/luisgen/node_modules/dotnet-2.0.0
> node -e "/*PostInstall: Installs platform-specific .NET framework */try{require('./dist/app.js')}catch(e){}"

events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: spawn /usr/local/bin/node ENOENT
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! dotnet-2.0.0@1.4.4 postinstall: `node -e "/*PostInstall: Installs platform-specific .NET framework */try{require('./dist/app.js')}catch(e){}"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the dotnet-2.0.0@1.4.4 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/vishwacsenakannan/.npm/_logs/2018-09-04T21_37_51_447Z-debug.log
```

# Bot Dispatch

```bash
> sudo npm config set registry https://botbuilder.myget.org/F/botbuilder-tools-daily/npm/

> sudo npm i -g botdispatch
/usr/local/bin/dispatch -> /usr/local/lib/node_modules/botdispatch/bin/dispatch.js

> dotnet-2.0.0@1.4.4 postinstall /usr/local/lib/node_modules/botdispatch/node_modules/dotnet-2.0.0
> node -e "/*PostInstall: Installs platform-specific .NET framework */try{require('./dist/app.js')}catch(e){}"

events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: spawn /usr/local/bin/node ENOENT
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! dotnet-2.0.0@1.4.4 postinstall: `node -e "/*PostInstall: Installs platform-specific .NET framework */try{require('./dist/app.js')}catch(e){}"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the dotnet-2.0.0@1.4.4 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/vishwacsenakannan/.npm/_logs/2018-09-04T21_39_20_426Z-debug.log
```

**Comment by enzocanoo** [comment]
Regarding `luis-apis`, the error was caused by an issue with the publish script fixed with #412.

For `luisgen` and `botdispatch` tools, the error seems to be related with `dotnet-2.0.0` dependency. You will see the same error if you run `node npm i -g dotnet-2.0.0`.
As a workaround, you could run `npm i -g dotnet-2.0.0` without `sudo` before installing `luisgen` or `botdispach`, and the tools will work fine.
